scif_app_recipes/ndcctools_v7.1.2_ubuntu16.04.scif

```diff
@@ -1,69 +1,78 @@
-%appinstall makeflow
+%appinstall odm_workflow
     # Install Makeflow
     conda create --prefix "${PWD}/conda" --yes -c conda-forge ndcctools
 
-%apprun makeflow
-    bash -c 'cp -r /input/* /scif/data/odm/images/ \
-    && conda run --prefix "${SCIF_APPROOT}/conda" makeflow \
+%apprun odm_workflow
+    bash -c 'cp -r /input/* /scif/data/odm/images/'
+    echo "{\"ORTHOMOSAIC_NAME\": \"${1}\", \"OUTPUT_#FOLDER\": \"/output\" }" > ${SCIF_APPDATA}/jx-params.json
+    ${SCIF_APPROOT_odm_workflow}/src/generate_geojson.sh "${2}"
+    bash -c 'conda run --prefix "${SCIF_APPROOT}/conda" makeflow \
         --jx \
-        --jx-args="${SCIF_APPROOT}/src/jx-args.json" \
+        --jx-args="/scif/apps/odm_workflow/src/jx-args.json" \
+        --jx-args="${SCIF_APPDATA}/jx-params.json" \
         --log-verbose \
         --retry-count=1 \
         --change-directory="${SCIF_APPDATA}" \
         --makeflow-log="${SCIF_APPDATA}/workflow.jx.makeflowlog" \
         --batch-log="${SCIF_APPDATA}/workflow.jx.batchlog" \
-        "${SCIF_APPROOT}/src/workflow.jx" \
-    && [ ! -f "/scif/data/soilmask/canopycover_fileslist.json" ]; "${SCIF_APPROOT}/src/prep-canopy-cover.sh" \
-    && conda run --prefix "${SCIF_APPROOT}/conda" makeflow \
+        "/scif/apps/odm_workflow/src/short_workflow.jx"'
+    [ ! -f "/output/canopycover_fileslist.json" ]; /scif/apps/makeflow/src/prep-canopy-cover.sh "${1}"
```

Comment:
I think this `prep-canopy-cover.sh` step is doing something that Makeflow can do by itself using the `listdir` function in the beginning of the `canopy-cover.jx` workflow. Example:

https://github.com/cooperative-computing-lab/makeflow-examples/blob/cf99ff99e443273b3a27a7ea51f52c6ce0df05e2/lifemapper/lifemapper.jx#L7

P.S. All the built-in functions: https://github.com/cooperative-computing-lab/cctools/blob/14cd27427d4fdb64f836c200cd9dd64d1f57d450/dttools/src/jx_function.h

Reply:
I will look into this. I haven't found anything like that yet.

Reply:
@julianpistorius The script is recursive. This is something that I haven't found Makeflow to be good at. Is there an example you can point me to?

Reply:
No, I don't know of a recursive example.